src/test/java/com/atlassian/db/replica/api/TestDualConnection.java

```diff
@@ -1791,4 +1792,39 @@ public void shouldPropagateLastWrite() throws SQLException {
 
         assertThat(lastWriteCache.get()).isPresent();
     }
+
+    @Test
+    public void shouldNotCallDirtyConnectionHookAfterARollback() throws SQLException {
+        final ConnectionProviderMock provider = getProviderThrowingExceptionsOnConnectionRollback();
+        final DirtyConnectionCloseHook dirtyConnectionCloseHook = mock(DirtyConnectionCloseHook.class);
+        final Connection connection = DualConnection.builder(provider, permanentInconsistency().build())
```

Comment:
`permanentInconsistency()` does it matter?

Reply:
No